tests/test_updater_rework.py

```diff
@@ -123,14 +123,13 @@ def setUp(self):
     # directory copied from the original repository files.
     tuf.settings.repositories_directory = self.client_directory
 
-    self.repository_mirrors = {'mirror1': {'url_prefix': url_prefix,
-                                           'metadata_path': 'metadata',
-                                           'targets_path': 'targets'}}
-
+    metadata_url = os.path.join(url_prefix, 'metadata/')
+    targets_url = os.path.join(url_prefix, 'targets/')
```

Comment:
Nit: should this be `urllib.parse.urljoin()`?

Reply:
yeah, I'll change that 

Fact is that the test needs to be rewritten, it's copied from test_updater.py and is horribly complex and full of issues like this for (I assume) Historical Reasons... but I'm trying to limit scope so not fixing anything else here